litmus-portal/graphql-server/pkg/chaoshub/chaoshub.go

```diff
@@ -515,31 +513,37 @@ func DeleteChaosHub(ctx context.Context, hubID string, projectID string) (bool,
 }
 
 // GetIconHandler ...
-var GetIconHandler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
-	vars := mux.Vars(r)
-	var img *os.File
-	var err error
-	var responseStatusCode int
-	if strings.ToLower(vars["ChartName"]) == "predefined" {
-		img, err = os.Open("/tmp/version/" + vars["ProjectID"] + "/" + vars["HubName"] + "/workflows/icons/" + vars["IconName"])
+var GetIconHandler = gin.HandlerFunc(func(c *gin.Context) {
+	var (
+		projectID          = c.Param("ProjectID")
+		hubName            = c.Param("HubName")
+		chartName          = c.Param("ChartName")
+		iconName           = c.Param("IconName")
+		img                *os.File
+		err                error
+		responseStatusCode int
+	)
+
+	if strings.ToLower(chartName) == "predefined" {
+		img, err = os.Open("/tmp/version/" + projectID + "/" + hubName + "/workflows/icons/" + iconName)
 		responseStatusCode = 200
 		if err != nil {
 			responseStatusCode = 500
-			fmt.Fprint(w, "icon cannot be fetched, err : "+err.Error())
+			fmt.Fprint(c.Writer, "icon cannot be fetched, err : "+err.Error())
```

Comment:
The code can be optimized to do the error check for both cases only once outside as the err is shared.

Reply:
Rest looks good!

Reply:
Delete duplicated codes in  [new commit](https://github.com/litmuschaos/litmus/pull/3923/commits/26a1619f6b6156a7a08ab52f3ef31f0790eee08f)